client/rest/test/plugins/metadata/metadataRoutes_spec.js

```diff
@@ -240,4 +240,67 @@ describe('metadata routes', () => {
 			});
 		});
 	});
+
+	describe('metal', () => {
+		const mockServer = new MockServer();
+		const db = {
+			binDataByMetalId: sinon.stub().resolves({ payload: 'db_payload', text: 'db_text' })
+		};
+		metadataRoutes.register(mockServer.server, db, {});
+
+		beforeEach(() => {
+			mockServer.resetStats();
+			mockServer.res.setHeader = sinon.spy();
+			mockServer.res.write = sinon.spy();
+			mockServer.res.end = sinon.spy();
+		});
+
+		describe('get by metal id', () => {
+			const callRouteAndAssert = async (route, req, shouldCallDb) => {
+				await mockServer.callRoute(route, req).then(() => {
+					expect(mockServer.res.write.calledOnce).to.equal(true);
+					expect(mockServer.next.calledOnce).to.equal(true);
+					expect(mockServer.res.setHeader.calledWithExactly('content-type', 'image/png')).to.equal(true);
+					expect(mockServer.res.setHeader
+						.calledWithExactly('Content-Disposition', 'attachment; filename="image.png"')).to.equal(true);
+					expect(mockServer.res.setHeader.calledWithExactly('Content-MetalText', 'db_text')).to.equal(true);
+					expect(mockServer.res.setHeader.calledWithExactly('Content-MetalText', 'db_fail_text')).to.equal(false);
+					expect(db.binDataByMetalId.calledOnce).to.equal(shouldCallDb);
+					if (shouldCallDb)
+						expect(db.binDataByMetalId.alwaysCalledWith('metal_id')).to.equal(true);
+				});
+			};
+
+			// Arrange:
+			const route = mockServer.getRoute('/metadata/metal/:metalId').get();
+			const req = {
+				params: {
+					metalId: 'metal_id',
+					mimeType: 'image/png',
+					fileName: 'image.png',
+					download: 'true'
+				}
+			};
+
+			it('returns page with results and uses DB on single call', async () => {
+				// Act & Assert:
+				callRouteAndAssert(route, req, true);
+			});
+
+			it('returns page with results and uses cache on second call', async () => {
+				// Arrange:
+				// First call
+				await mockServer.callRoute(route, req);
+
+				// Reset the spies
+				mockServer.res.write.resetHistory();
+				mockServer.next.resetHistory();
+				mockServer.res.setHeader.resetHistory();
+				db.binDataByMetalId.resetHistory();
+
+				// Act & Assert:
```

Comment:
Generally we've been using '+' instead of '&'

Reply:
yeees!!! i knew it!!!
why i used '&'😂